design_document.md

```diff
@@ -0,0 +1,172 @@
+
+
+    COINSTACK {
+    MAP<COIN,int> {
+        COIN(1): 100,
+        COIN(2): 100,
+        COIN(5): 100,
+        COIN(10): 100,
+        COIN(20): 100,
+        COIN(50): 100,
+        COIN(100): 100,
+        COIN(200): 100,
+        }
+
+        get sum of all coin
+    }
+
+    COIN {
+        constuctor(required this.cent)
+        int cent
+    }
+
+```
+___
+
+
+
+
+<span id=insert_money_window></id>
+
+___
+#  GUI:
+
+## sidebar
+
+#### coin slot
+a visual widget with no function
+
+#### bill slit
+a visual widget with no function
+
+#### inputfield
+a visual widget with no function.
+
+#### insert money button
+is a button obove the return money button.
+on tap:  <a href=#open_insert_money_window> openInsertMoneyWindow()</a>
+<span id= return_money_button></id>
+
+#### return money button
+is placed on the <a href = #sidebar>sidebar</a> obove <a href=#input_field>input field</a><br>
+ on tap: <a href=#return_money>return money()</a>
+ if money is returned into the <a href = #coin_tray>coin tray</a> it becomes clickable.
+
+<span id = coin_tray></id>
+
+#### coin tray
+the coin tray is a container widget which when the user has returned his money or bought an item and there is exchange money left becomes clickable. on tap: <a href=#return_money> return money() </a>. If the user does not take out the money from the coin tray and gets more money exchanged i.e after another transaction, then the new exchange money is added onto the old exchange money.
+<span id = item_slot></id>
+
+## product area
+
+<span id=item_slot></id>
+
+#### product slot
+is a clickable Container widget.Has a column child with 1. its product number 2. product picture (2d icon) 3. product price.
+ on tap: <a href=#buyItem> buyItem()</a>
+
+
+
+## dispension area
+<span id = dispensing_slot></id>
+
+#### dispensing slot
+is below the sidebar and the product area which becomes clickable after a product has been bought.
+on tap:
+<a href=#takeOutItem> takeOutItem() </a>
+
+
+### conditional
+
+#### admin password window
+a AltertDialogue and a numberfield appears which asks the user to input a password. The user can close the window which resets the inputfield or confirm. If the password is correct the user will enter the admin mode. If it is wrong he will be notified "wrong password"
+
+#### insert money window
+a AlertDialogue and a number field appears up where the user sees all availabe € coins. on tapping any coin it will safe the coin in a temp account. The user has the option to close the window which resets the textfield. 
+Or press the confirm button which then returns the amount of each coin and adds them to the purchase total and resets the textfield.
+___
+
+
+## Database
+
+## Admin
+the admin mode is accessed by long pressing the admin button which opens a number field expecting a correct password. The user will enter admin mode after inserting the correct password and confirming.
+Password: "0000"
+
+## functionalities
+
+#### insertCoin(COIN)
```

Comment:
If this is a sub-section of *functionalities* which has two `#`s, use three `#`s in front of the sub-heading texts. Same goes for the other sub-headings.